commands/push.js

```diff
@@ -1,77 +1,82 @@
+  ${ cli.color.white('heroku container:push web worker --recursive')}     # Pushes Dockerfile.web and Dockerfile.worker found in current & subdirectories
+  ${ cli.color.white('heroku container:push --recursive')}                # Pushes Dockerfile.* found in current & subdirectories`
 
-module.exports = function(topic) {
+module.exports = function (topic) {
   return {
     topic: topic,
     command: 'push',
-    description: 'Builds, then pushes a Docker image to deploy your Heroku app',
+    description: 'builds, then pushes Docker images to deploy your Heroku app',
     needsApp: true,
     needsAuth: true,
-    args: [{ name: 'process', optional: true }],
-    flags: [{ name: 'verbose', char: 'v', hasValue: false }],
-    run: cli.command(co.wrap(push))
-  };
-};
+    variableArgs: true,
+    help: usage,
+    flags: [
+      {
+        name: 'verbose',
+        char: 'v',
+        hasValue: false
+      },
+      {
+        name: 'recursive',
+        char: 'R',
+        hasValue: false
+      }
+    ],
+    run: cli.command(push)
+  }
+}
 
-function* push(context, heroku) {
-  let herokuHost = process.env.HEROKU_HOST || 'heroku.com';
-  let registry = `registry.${ herokuHost }`;
-  let proc = context.args.process || 'web';
-  let resource = `${ registry }/${ context.app }/${ proc }`;
+let push = async function (context, heroku) {
+  const recurse = !!context.flags.recursive
+  if (context.args.length === 0 && !recurse) {
+    cli.error(`Error: Requires either --recursive or one or more process types\n ${usage} `)
+    process.exit(1)
+  }
+  if(context.args.length > 1 && !recurse){
+    cli.error(`Error: Please specify one target process type\n ${usage} `)
+    process.exit(1)
+  }
+  let herokuHost = process.env.HEROKU_HOST || 'heroku.com'
+  let registry = `registry.${ herokuHost }`
+  let dockerfiles = Sanbashi.getDockerfiles(process.cwd(), recurse)
+  let processTypes = ['standard']
+  if(recurse){
+    processTypes = context.args
+  }
+  let possibleJobs = Sanbashi.getJobs(`${ registry }/${ context.app }`, processTypes, dockerfiles)
+  let jobs = await Sanbashi.chooseJobs(possibleJobs)
+  if (!jobs.length) {
+    cli.warn('No images to push')
+    process.exit(1)
+  }
 
   try {
-    let build = yield buildImage(resource, context.cwd, context.flags.verbose);
+    for (let job of jobs) {
+      cli.styledHeader(`Building ${job.name} (${job.dockerfile})`)
+      await Sanbashi.buildImage(job.dockerfile, job.resource, context.flags.verbose)
+    }
   }
   catch (err) {
-    cli.error(`Error: docker build exited with ${ err }`);
-    process.exit(1);
+    cli.error(`Error: docker build exited with ${ err }`)
+    cli.hush(err.stack || err)
+    process.exit(1)
   }
 
   try {
-    let push = yield pushImage(resource, context.flags.verbose);
+    for (let job of jobs) {
+     cli.styledHeader(`Pushing  ${job.name}  (${job.dockerfile })`)
```

Comment:
there are a lot of little style issues like the indentation on this line. It's probably too much work to put it in for the whole project, but it would be good to run it at least on the new code